dev/make-distribution.sh

```diff
@@ -42,6 +42,7 @@ function exit_with_usage {
   echo "usage:"
   cl_options="[--name] [--tgz] [--mvn <mvn-command>]"
   echo "make-distribution.sh $cl_options <maven build options>"
+  echo "Decrease build times with the -T option: -T 1C will result in Maven using one thread per core."
```

Comment:
I should have been more clear, the idea was to document **if you specify a Maven command** you can pass -T 1C to that Maven command, but I like what you tried and think this is how it should be implemented so I'm adding this feature and testing with:

dev/make-distribution.sh --name hadoop-2.7.0 -T 1C -Psparkr -Pyarn -Phadoop-2.7 -Phive -Phive-thriftserver -Dscala-2.11

Will update my pull request shortly, almost done